### PR TITLE
[transfer] Stop JSON marshaling/unmarshaling event payloads

### PR DESCRIPTION
**Pull request by nathan-artie**
Transfer can now handle `int16`/`int32`/`int64` values for integer types, `int64` values and time types, and `[]byte` values for decimal types, so we don't have to marshal and unmarshal each row to JSON anymore.

### CODE REVIEW

File: writers/transfer/writer.go [hunk 1]
```diff
@@ -85,11 +50,7 @@ func (w *Writer) Write(_ context.Context, messages []lib.RawMessage) error {
 
 	var events []event.Event
 	for _, message := range messages {
-		evt, err := messageToMemoryEvent(message, w.tc)
-		if err != nil {
-			return err
-		}
-		events = append(events, evt)
+		events = append(events, event.ToMemoryEvent(message.Event(), message.PartitionKey(), w.tc, config.Replication))
```
Comment (Tang8330) [inline writers/transfer/writer.go:53]: Didn't JSON marshal also base64 encode? 

Is that covered too?

Comment (nathan-artie) [reply writers/transfer/writer.go:53]: It is for `"org.apache.kafka.connect.data.Decimal"` and `"io.debezium.data.VariableScaleDecimal"`

Comment (nathan-artie) [reply writers/transfer/writer.go:53]: Though we probably should handle it for plain `[]byte` values too.

Comment (Tang8330) [reply writers/transfer/writer.go:53]: Not sure if I follow - can you elaborate?

Comment (nathan-artie) [reply writers/transfer/writer.go:53]: https://github.com/artie-labs/transfer/pull/417

Comment (Tang8330) [reply writers/transfer/writer.go:53]: Makes sense

**Review by Tang8330 — APPROVED** [review]

**Pull request merged**